jaxrs/WFLY-14559-new-attributes.adoc

```diff
@@ -0,0 +1,141 @@
+every deployment.
+
+== Issue Metadata
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14559[WFLY-14559]
+* https://issues.redhat.com/browse/RESTEASY-2646[RESTEASY-2646]
+* https://issues.redhat.com/browse/RESTEASY-2280[RESTEASY-2280]
+* https://issues.redhat.com/browse/RESTEASY-2782[RESTEASY-2782]
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+The following attributes should be added to the `jaxrs` subsystem:
+
+* https://issues.redhat.com/browse/RESTEASY-2646[RESTEASY-2646] `resteasy-match-cache-enabled`: Defaults to `true` and can be disabled to no longer cache resource requests to the same URL.
+* https://issues.redhat.com/browse/RESTEASY-2646[RESTEASY-2646] `resteasy-match-cache-size` : Defaults to `2048` and when the cache is enabled, changes the size limit for the cache.
+* https://issues.redhat.com/browse/RESTEASY-2280[RESTEASY-2280] `resteasy-patchfilter-disabled` : Defaults to `false`. Disables the filter used for `application/merge-patch+json` requests if set to `true`.
+* https://issues.redhat.com/browse/RESTEASY-2782[RESTEASY-2782] `resteasy-original-webapplicationexception-behavior` : Defaults to `false`. Set to `true`, this parameter will restore the original behavior in which a Client running in a resource method will throw a Jakarta RESTful Web Services `WebApplicationException` instead of a RESTEasy version with a sanitized Response.
+
+These attributes will be added with their respective defaults. The value of the attributes will be used to set implicit
+servlet context parameters on each deployment. These will only be set if the context parameter was not already set as
+a servlet context parameter by the application being deployed.
+
+=== Non-Requirements
+
+N/A
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+There should be no backwards compatibility issues. These options are already available to be set via other means.
+
+=== Default Configuration
+
+These are new attributes, but given they will all have defaults they should not change the current configuration.
+
+=== Importing Existing Configuration
+
+This will have no effect on existing configurations given these are new attributes with default values.
+
+=== Deployments
+
+Deployments should not be affected by this change.
+
+=== Interoperability
+
+N/A
+
+== Implementation Plan
+
+The implementation will be to simply add the new attributes to the subsystem. In a `DeploymentUnitProcessor`, which
+already exists, the attributes will be converted into context properties set in the servlet context parameters. The
+values from the subsystem will not overwrite values defined in the `web.xml` of the deployment being processed.
```

Comment:
@jamezp if one of the new attributes is not set in the model, would or wouldn't it be added to the deployment context parameters ... I mean, would it or wouldn't it influence whatever default is already set in resteasy?

I am asking this because so far, we had:

- resteasy default values

NExt we'll have:

- resteasy default values 
- wf default values 

in case the 2 default values diverge any time in some future, we might have "wf default != resteasy default" resulting in a dis-alignment ... I guess I am just asking how we deal with such things with existing attributes